cli/src/commands/branch.rs

```diff
@@ -297,6 +297,13 @@ fn cmd_branch_set(
     let target_commit =
         workspace_command.resolve_single_rev(args.revision.as_deref().unwrap_or("@"), ui)?;
     let repo = workspace_command.repo().as_ref();
+    let is_fast_forward = |old_target: &RefTarget| {
+        // Strictly speaking, "all" old targets should be ancestors, but we allow
```

Comment:
This comment doesn't seem appropriate any more, since this function only checks all the ancestors.

Could you move/make a comment explaining why conflicted branches can still be set without `-B`? I'm a bit tired and couldn't figure it out; if I missed something really obvious, feel free to do something else.

Finally, do we have a test checking this case (setting a conflicted branch to one of the conflicted possibilities without `-B`)? Optionally, you could make one.

Reply:
I'm realizing that a possible reason for my confusion is that I misunderstood how the whole business with setting conflicted commits works. I'm not going to look at this any more today, though. Feel free to wait or ask for another review, or to just resolve this discussion if it isn't helpful.

Reply:
The logic is unchanged. It's still allowed to move branch to descendant of one of the old conflicting targets, though it's not strictly a fast-forwardable move.

Anyway, adding tests makes sense. Done.